Dockerfile

```diff
@@ -1,12 +1,22 @@
-FROM openjdk:8u242-jre
+FROM ethereum/solc:0.4.24 as builder
+# We use a solidity container to compile our contracts
 
+# All our files are transfered to a single folder called "app"
 run mkdir /app
 COPY target/lightchain-container-*-jar-with-dependencies.jar /app
-# COPY lightchain-container/src/main/resources/log4j.properties /app
+COPY src/main/resources/log4j.properties /app
 COPY simulation.config /app
+COPY contracts/*.sol /app/
+WORKDIR /app
+# NOTE: I hard coded the filename here (eg. testcon.sol)
+RUN  solc -o . --bin testcon.sol
 
+# Here we use a java container for further simulation
+FROM openjdk:8u242-jre
 WORKDIR /app
+# The files from above solidity container are copied into this current container
+COPY --from=builder /app .
 
 # NOTE: I hard coded the version here
 # TODO: Make the filename below change dynamically with new versions
-CMD ["java", "-cp", "LightChain-0.0.1-SNAPSHOT-jar-with-dependencies.jar", "simulation.SimulationDriver"]
+CMD ["java", "-cp", "lightchain-container-0.0.1-SNAPSHOT-jar-with-dependencies.jar", "simulation.SimulationDriver"]
```

Comment:
does it mean that we should build the jar manually? if that is the case, please add a building the jar in the makefile (i.e., `make container-build` should build the jar and then run this docker file)

Reply:
No its is used to run the jar file. This was a bug in the previously commented code.